### PR TITLE
fix(span-tree): Autogrouped sibling spans bugs

### PR DESCRIPTION
**Pull request by 0Calories**
This PR fixes several bugs associated with the autogrouped sibling spans in the span tree:

- [x] Duration text gets messed up when you narrow the view in the span view
- [x] Autogrouped sibling spans don’t show up on the minimap
- [x] Can’t search by ID for individual spans within expanded sibling span groups
- [x] Show description as well as op in the autogrouped sibling spans, instead of just the op
- [x] Autogrouped sibling spans should properly by marked as out of view when they are not in highlighted in the minimap

### Also includes some slight refactors:

- [x] Make a separate component for autogrouped sibling spans vs autogrouped descendants
- [x] Move `GroupType` into `types.tsx`

### Bugs that still need to be addressed (Will come in a follow up PR)
- [ ] Sibling groups can have the same op and description, this causes separate groups to get expanded/collapsed at the same time

Fixes PERF-1499

### CODE REVIEW

File: static/app/components/events/interfaces/spans/spanTreeModel.tsx [hunk 1]
```diff
@@ -415,6 +415,73 @@ class SpanTreeModel {
...
+            group.forEach(spanModel =>
+              acc.descendants.push({
+                type: 'filtered_out',
+                span: spanModel.span,
+              })
+            );
+            return acc;
+          }
+
+          // Check if the sibling group is out of bounds
```
Comment (k-fish) [inline static/app/components/events/interfaces/spans/spanTreeModel.tsx:468]: nit: ditto above re: `what` style

File: static/app/components/events/interfaces/spans/spanTreeModel.tsx [hunk 2]
```diff
@@ -415,6 +415,73 @@ class SpanTreeModel {
 
           // This may not be the case, and needs to be looked into later
 
+          const key = getSiblingGroupKey(group[0].span);
+          if (this.expandedSiblingGroups.has(key)) {
+            // Check if any of the spans in the sibling group are filtered out (possible when filtering by span ID)
+            group.forEach((spanModel, index) => {
+              if (this.isSpanFilteredOut(props, spanModel)) {
```
Comment (k-fish) [inline static/app/components/events/interfaces/spans/spanTreeModel.tsx:422]: Maybe expand on why/how the parent can be unfiltered but the children can be filtered? Or is this an issue of `return` ordering? If you move the filter condition below above this one, does it affect how it's display? If we're filtering out the parent do we still show children?

**Review by k-fish — COMMENTED** [review]
Nice! Main blocker for approve is that if you want to have two separate files when the functionality is so close we should prob at least split all of it out into reusable functions so we don't have issues down the line with divergence. The nits re: commenting are minor, fix them if you want.

File: static/app/components/events/interfaces/spans/spanGroupBar.tsx [hunk 3]
```diff
@@ -53,104 +53,94 @@ import {
...
   treeDepth: number;
-  toggleSiblingSpanGroup?: (span: SpanType) => void;
 };
 
-function getSpanGroupTimestamps(spanGroup: EnhancedSpan[]) {
-  return spanGroup.reduce(
-    (acc, spanGroupItem) => {
-      const {start_timestamp, timestamp} = spanGroupItem.span;
+class SpanGroupBar extends React.Component<Props> {
+  getSpanGroupTimestamps(spanGroup: EnhancedSpan[]) {
```
[the remark this answers is not in the document]
Comment (0Calories) [reply static/app/components/events/interfaces/spans/spanGroupBar.tsx:70]: Good catch, I forgot to do this 👍

File: static/app/components/events/interfaces/spans/spanSiblingGroupBar.tsx [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+  generateBounds: (bounds: SpanBoundsType) => SpanGeneratedBoundsType;
+  isLastSibling: boolean;
+  span: Readonly<ProcessedSpanType>;
+  spanGrouping: EnhancedSpan[];
+  spanNumber: number;
+  treeDepth: number;
+  toggleSiblingSpanGroup?: (span: SpanType) => void;
+};
+
+export default function SpanSiblingGroupBar(props: Props) {
```
[the remark this answers is not in the document]
Comment (0Calories) [reply static/app/components/events/interfaces/spans/spanSiblingGroupBar.tsx:62]: True that's a good point, I'll make some functions for this

[the remark this answers is not in the document]
Comment (0Calories) [reply static/app/components/events/interfaces/spans/spanSiblingGroupBar.tsx:62]: Hey @k-fish , I've decided I'm going to use composition for this. I'll refactor and make a generic `SpanGroupBar`  component (the current `SpanGroupBar` will be renamed to `SpanDescendantGroupBar`) that will handle rendering the common factors between these two. Currently working on this, but the refactor is going to make the PR a bit harder to read, so I'll open up a separate PR for this.

File: static/app/components/events/interfaces/spans/spanTreeModel.tsx [hunk 5]
```diff
@@ -415,6 +415,73 @@ class SpanTreeModel {
...
+
+                acc.previousSiblingEndTimestamp = spanModel.span.timestamp;
+                acc.descendants.push(enhancedSibling);
+              }
+            });
+
+            return acc;
+          }
+
+          // Check if the sibling group is filtered out
```
[the remark this answers is not in the document]
Comment (0Calories) [reply static/app/components/events/interfaces/spans/spanTreeModel.tsx:457]: You're right, this is a habit I have to work towards unlearning haha

**Comment by 0Calories** [comment]
Going to open up a separate PR in a bit, since I am making a refactor that will make `SpanGroupBar` a common component, with `SpanDescendantGroupBar` and `SpanSiblingGroupBar` as specialized components.

**Review by k-fish — APPROVED** [review]
Yeah those snapshots look better 👍

**Pull request merged**